raft.go

```diff
@@ -411,6 +411,7 @@ func (r *Raft) Shutdown() Future {
 	if !r.shutdown {
 		close(r.shutdownCh)
 		r.shutdown = true
+		r.trans.Close()
```

Comment:
Could you instead to a two-value type assertion `closeable, ok := r.trans.(closeableInterface)` and close if it worked? That would alleviate the need to change the exported interface.

If you do this, please update the appropriate comments/documentation to reflect "transport will be closed in x circumstances if closeable"